CoffeeChat/View/PausePearView.swift

```diff
@@ -0,0 +1,125 @@
+        oneWeekButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
+        addSubview(oneWeekButton)
+        
+        twoWeekButton.setTitle("2 Weeks", for: .normal)
+        setButtonAppearance(button: twoWeekButton)
+        twoWeekButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
+        addSubview(twoWeekButton)
+        
+        threeWeekButton.setTitle("3 Weeks", for: .normal)
+        setButtonAppearance(button: threeWeekButton)
+        threeWeekButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
+        addSubview(threeWeekButton)
+        
+        cancelButton.setTitle("Cancel", for: .normal)
+        cancelButton.setTitleColor(.textGreen, for: .normal)
+        cancelButton.backgroundColor = .clear
+        cancelButton.titleLabel?.font = ._16CircularStdBook
+        cancelButton.addTarget(self, action: #selector(cancelPause), for: .touchUpInside)
+        addSubview(cancelButton)
+    }
+
+    private func setButtonAppearance(button: UIButton) {
+        button.layer.cornerRadius = 8
+        button.backgroundColor = .white
+        button.setTitleColor(.black, for: .normal)
+        button.titleLabel?.font = ._16CircularStdBook
+    }
+
+    
+    private func setupConstraints() {
+        let weekButtonSize = CGSize(width: 110, height: 40)
+        let cancelButtonSize = CGSize(width: 52, height: 20)
+        
+        pauseLabel.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().offset(36)
+            make.leading.trailing.equalToSuperview().inset(75)
+        }
+        
+        oneWeekButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(pauseLabel.snp.bottom).offset(22)
+            make.size.equalTo(weekButtonSize)
+        }
+        
+        twoWeekButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(oneWeekButton.snp.bottom).offset(12)
+            make.size.equalTo(weekButtonSize)
+        }
+        
+        threeWeekButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(twoWeekButton.snp.bottom).offset(12)
+            make.size.equalTo(weekButtonSize)
+        }
+        
+        cancelButton.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(threeWeekButton.snp.bottom).offset(30)
+            make.size.equalTo(cancelButtonSize)
+        }
+    }
+    
+    @objc private func buttonAction(button: UIButton) {
+        if let pauseTime = button.titleLabel?.text {
+            delegate?.pausePearAction(data: pauseTime)
+            oneWeekButton.backgroundColor = .white
+            twoWeekButton.backgroundColor = .white
+            threeWeekButton.backgroundColor = .white
+            button.backgroundColor = .pearGreen
```

Comment:
I think it might be good to have a class variable for what the pause state is currently? So like the variable can say the user is currently set to be paused for "2 weeks" so then that is the button that is colored and when another button is clicked, the variable gets set to the appropriate amount. That may be a bit more efficient than assigning each button's color here. Also cause we need that value anyways, for when we click save.

Reply:
how would the other buttons' background color get set to white?